### PR TITLE
Allows specification of processes per block and force decomp via map argument

### PR DESCRIPTION
**Pull request by reguly**
see mblock app, run with extra argument "demo"

### CODE REVIEW

File: ops/c/src/mpi/ops_mpi_partition.cpp [hunk 1]
```diff
@@ -171,8 +171,8 @@ void ops_partition_blocks(int **processes, int **proc_offsets, int **proc_disps,
       //Sanity check for force decomp
       int prod = 1;
       for (int d = 0; d < ndim; d++) prod *= (pdims[d]==0 ? 1 : pdims[d]);
-      if (prod > processes_per_block[i]) throw OPSException(OPS_RUNTIME_CONFIGURATION_ERROR, "Error: force_decomp requested more processes than available for the block");
-      
+      if (prod != processes_per_block[i]) throw OPSException(OPS_RUNTIME_CONFIGURATION_ERROR, "Error: force_decomp requested processes inconsistent with one specified for the block");
```
Comment (reguly) [inline ops/c/src/mpi/ops_mpi_partition.cpp:174]: That is not right - you can specify 0 if you don't care about the number of processes in that direction. So prod needs to be less or equal, so MPI_Dims_create will work.

Comment (jpmeng) [reply ops/c/src/mpi/ops_mpi_partition.cpp:174]: Ok, See. Can we only pick up the right one here?

Comment (reguly) [reply ops/c/src/mpi/ops_mpi_partition.cpp:174]: I'm not sure what you mean. My original check is good for the case where some decomp sizes are not specified

Comment (jpmeng) [reply ops/c/src/mpi/ops_mpi_partition.cpp:174]: I mean if we can pick up your comment to merge and ignore my wrong commit by using Github?

Comment (jpmeng) [reply ops/c/src/mpi/ops_mpi_partition.cpp]: I reverted it. please see blow.

**Comment by reguly** [comment]
Added support for specifying all this via the command line arguments OPS_FORCE_DECOMP_* and OPS_PROCESSES_PER_BLOCK, both can take a comma-separated list of arguments

**Comment by reguly** [comment]
@gihanmudalige @jpmeng this should be ready to merge, please test and review

**Comment by jpmeng** [comment]
> @gihanmudalige @jpmeng this should be ready to merge, please test and review

Sure. I will have a go tomorrow. Jianping

**Review by jpmeng — APPROVED** [review]

**Pull request merged**